bundles/catalogue/metadataflyout/view/MetadataPage.js

```diff
@@ -49,22 +49,28 @@ Oskari.clazz.define('Oskari.catalogue.bundle.metadataflyout.view.MetadataPage',
 
             template = _.extend({}, data);
             delete template.identifications;
-            // Create a panel for each identification
-            for (i = 0; i < data.identifications.length; i += 1) {
-                model = _.extend({}, template);
-                model.identification = data.identifications[i];
-                panel = Oskari.clazz.create(
-                    'Oskari.catalogue.bundle.metadataflyout.view.MetadataPanel',
-                    me.instance,
-                    me.locale,
-                    model
-                );
-                if (me.asyncTabs && !jQuery.isEmptyObject(me.asyncTabs)) {
-                    panel.addTabs(me.asyncTabs);
-                }
 
-                me.addPanel(panel);
-                panel.init(i === 0);
+            if (data.identifications.length === 0) {
+                //  No identifications, show metadata not found message
+                me._showMetadataNotFoundMessage();
+            } else {
+                // Create a panel for each identification
+                for (i = 0; i < data.identifications.length; i += 1) {
```

Comment:
Using `.forEach` for iterating an Array would be less error prone, because you don't need to keep track of the index. The function given to ´.forEach` is called with the index as the second argument, if the index is needed in the logic.